### PR TITLE
Migrate tests to use TestCoroutineScope.

### PR DESCRIPTION
**Pull request by Tapchicoma**
Migrate all unit tests in `:library:core` to use `TestCoroutineScope`, that executes coroutines immediately and allows to advance time manually. Such approach should make tests more reliable and reduce flakyness.

More info about `TestCoroutineScope` could be found here: https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-test/README.md

### CODE REVIEW

**Comment by befrvnk** [comment]
I need some more time to review this. Is it ok for you if I review it next week on Monday?

**Comment by Tapchicoma** [comment]
Yes, this is ok :slightly_smiling_face:

File: library/core/src/test/kotlin/com/freeletics/coredux/CancellableSideEffectTest.kt [hunk 1]
```diff
@@ -35,89 +37,75 @@ object CancellableSideEffectTest : Spek({
...
+        afterEach { testScope.cleanupTestCoroutines() }
+
         context("when current state is not \"\"") {
             beforeEach { stateAccessor.setCurrentState("some-other-state") }
 
             context("and on new 1 action") {
-                beforeEach { scope.launch { inputChannel.send(1) } }
+                beforeEach {
+                    testScope.launch { inputChannel.send(1) }
+                    testScope.advanceTimeBy(DEFAULT_DELAY_MS + 1)
```
Comment (befrvnk) [inline library/core/src/test/kotlin/com/freeletics/coredux/CancellableSideEffectTest.kt:55]: What would happen if you don't set the time here?

**Review by befrvnk — APPROVED** [review]
👍

File: library/core/src/test/kotlin/com/freeletics/coredux/CancellableSideEffectTest.kt (hunk 1 above)
Comment (Tapchicoma) [reply library/core/src/test/kotlin/com/freeletics/coredux/CancellableSideEffectTest.kt:55]: `delay(DEFAULT_DELAY_MS)` at: https://github.com/freeletics/CoRedux/blob/6980690e9cb376f883edfbac5cde379ad97b44d8/library/core/src/test/kotlin/com/freeletics/coredux/CancellableSideEffectTest.kt#L26 will not pass and side effect under test will not start emitting items.

**Pull request merged**